scene/resources/surface_tool.cpp

```diff
@@ -315,19 +315,16 @@ void SurfaceTool::set_uv2(const Vector2 &p_uv2) {
 	last_uv2 = p_uv2;
 }
 
-void SurfaceTool::set_custom(int p_index, const Color &p_custom) {
-	ERR_FAIL_INDEX(p_index, RS::ARRAY_CUSTOM_COUNT);
+void SurfaceTool::set_custom(int p_channel_index, const Color &p_custom) {
+	ERR_FAIL_INDEX(p_channel_index, RS::ARRAY_CUSTOM_COUNT);
 	ERR_FAIL_COND(!begun);
-	ERR_FAIL_COND(last_custom_format[p_index] == CUSTOM_MAX);
```

Comment:
I prefer to still enforce setting ``last_custom_format`` before setting the first vertex as it allows us to warn users when they are setting custom data without setting a format. 

With these changes, it is entirely possible that users will call ``set_custom`` but never call call ``set_custom_format`` in which case the custom data will be silently ignored. By forcing users to ``set_custom_format`` before adding data, we can ensure that they actually set the custom format. 

I don't see a reason why it is helpful for users to set the custom format after setting all the data anyway. In practice, you need to know your custom format before setting the data anyway, so requiring ``set_custom_format`` before the first call to ``set_vertex`` is not a big deal.

Reply:
Yeah good catch. I added the check back in. That's also how I wrote the documentation.